### ISSUE TITLE
[A11Y] Unable to scroll down the home page when magnification is turned on

### ISSUE THREAD

**Issue by KolliAnitha** [oppia/oppia-android#4318]
**Describe the bug**
Unable to scroll down the home page when magnification is turned on

Pre-Requisite: Turn on magnification mode

**To Reproduce**
1. Login to Oppia as Admin
2. Scroll down the page to view All Topics

**Expected behavior**
Users should be able to scroll down the page when magnification is turned on

Actual Result:
Unable to scroll down the page when magnification is turned on in both landscape mode and portrait mode

https://drive.google.com/file/d/1LJ93DIX_7S_-xaEgsO0NXzrir0iZ_2Z_/view?usp=drivesdk

**Environment**
Device name: One plus Nord2 5G
Android version : Android 11
App version : 0.7-alpha-d1b2827517

**Comment by BenHenning** [comment]
Interesting. My first thought is that this must have something to do with RecyclerView scrolling, but then I'd expect other apps to have the issue.

@KolliAnitha could you check apps like Google News (that have a list of items to scroll similar to Oppia's home screen) with magnification mode on the same device to see if it has the same problem? It might also be nice to get a video of what "correct" looks like here so that we have a point of reference.

**Comment by KolliAnitha** [comment]
@BenHenning, As per the standards to scroll down the page we should drag 2 fingers. Please check the link below.
https://support.google.com/accessibility/android/answer/6006949?hl=en
However if i drag with 1 finger I am able to scroll down the page and view all topics.

Even in Google News app it is the same behavior i.e on dragging with 1 finger I am able to scroll down and view the topics.

https://user-images.githubusercontent.com/101634267/165473346-1c7c702d-6877-4504-849d-b4154b843320.mp4

https://user-images.githubusercontent.com/101634267/165473550-99c65d1b-917d-435e-8bfe-9ee90c609049.mp4

**Comment by BenHenning** [comment]
That's helpful--thanks @KolliAnitha.

This actually seems potentially bad since it'll make it hard for TalkBack users to start a specific topic. Sending this over to the CLaM team for a11y triage.

**Comment by rt4914** [comment]
For me it is working very differently. This is how I used the magnification:

1. Turn on the magnification
2. Tap on the screen (This is needed to make sure that we are using permanent mode, if we do not tap and directly start using then magnification is in temporary mode in which case as soon as we lift of the finger the magnification turns off.)

_(At this point the magnification tool will just magnify the entire content of the current visible screen so if the screen is scrollable it won't show that content.)_

3. Now use two-finger scroll to move within the magnified screen.
4. Use one-finger scroll to move outside the magnified screen.

Have a look at this video:

https://user-images.githubusercontent.com/9396084/173010386-7caae75d-dd42-4f3a-b0be-3794108550e6.mp4


## Whatsapp experience

I believe whatsapp works in similar way. Have a look at this video where I have performed the above steps and it does work as expected.


https://user-images.githubusercontent.com/9396084/173012374-7b5f880c-bc1e-4512-b58c-74e803c1229b.mp4

**Comment by vrajdesai78** [comment]
I would agree with @rt4914. I have checked on my device and it works the same for me if I have to scroll within magnified content then I can use two fingers and if I want to scroll outside that I can use a single finger. This works the same for other applications also like google news. 

https://user-images.githubusercontent.com/43074241/173082896-41cc0b96-f491-4f12-a5d1-c68516095970.mp4

**Comment by BenHenning** [comment]
For searching context, this issue was found in 0.7-alpha (MR4).

**Comment by BenHenning** [comment]
Thanks @rt4914 and @vrajdesai78 for the context.

@KolliAnitha does two-finger scrolling work in your original repro conditions?

**Comment by KolliAnitha** [comment]
Hi @BenHenning, With two-finger scrolling i am unable to scroll down the page completely. Please refer to the below attachment.
https://user-images.githubusercontent.com/101634267/173391310-1cf7257b-8511-4123-9d51-e815c62dd5ec.mp4

**Comment by vrajdesai78** [comment]
@KolliAnitha according to my observation with two fingers you can just scroll to the current magnified area and to scroll outside it you have to scroll using one finger.

**Comment by BenHenning** [comment]
@KolliAnitha I think your video is showing the same behavior as @rt4914 described in his comment above, or am I missing an important part of the video? Specifically, that 2 fingers can be used to move within the zoomed region, and one finger can be used to move the region itself (thereby allowing scrolling).

**Comment by rt4914** [comment]
This is actually working correctly. To understand this correctly have a look at have a look at below video first:

https://user-images.githubusercontent.com/9396084/173012374-7b5f880c-bc1e-4512-b58c-74e803c1229b.mp4

Now if you notice that in the zoomed-in region I had to use `two-fingers` to navigate but to enter the zoomed-out region I had to use `one-finger` swipe and this is the expected behaviour. 
This is exactly same for Google News as well as Oppia app.

@KolliAnitha Can you please confirm if it clears your doubt?

**Comment by KolliAnitha** [comment]
@rt4914, got your point, I think we can close this issue.
